### ISSUE TITLE
Change Request: Move the `language` option to `settings.language`.

### ISSUE THREAD

**Issue by m1rn** [eslint/markdown#277]
### Environment

Node version: v22.1.0
npm version: v10.7.0
Local ESLint version: v9.9.1 (Currently used)
Global ESLint version: Not found
Operating System: win32 10.0.22631

### What problem do you want to solve?

`@eslint/markdown` and `@eslint/json` both have a `language` option that needs to be defined within the ESLint configuration object. However, this property is not actually documented in the ESLint configuration object documentation, leading to editor complaints when using TypeScript. I would prefer that such custom fields be defined in `settings`, for example:

```js
import markdown from '@eslint/markdown';

export default [
  {
    files: ['**/*.md'],
    plugins: {
      markdown,
    },
    rules: {
      'markdown/fenced-code-language': 2
    },
    processor: 'markdown/markdown',
    settings: {
      language: 'markdown/commonmark',
    }
  }
];
```

### What do you think is the correct solution?

-

### Participation

- [ ] I am willing to submit a pull request for this change.
